File: 백준/Gold/3079. 입국심사/입국심사.py
import sys
input=sys.stdin.readline

N,M=map(int,input().split())
l=[]
for i in range(N):
  l.append(int(input()))

start=min(l)
end=max(l)*M
answer=end
while start<=end:
  mid=(start+end)//2
  cnt=0
  for i in range(N):
    cnt+=mid//l[i]

  if cnt>=M:
    if answer>mid:
      answer=mid
    end=mid-1
  else:
    start=mid+1

print(answer)

File: 백준/Gold/3079. 입국심사/입국심사.js
const readline = require('readline');
const rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout,
});

const input = [];
let iter = 0;
rl.on('line', (line) => {
  input.push(line);
}).on('close', () => {
  const [N, M] = [...input[iter++].split(' ')].map((item) => Number(item));

  const time = [];
  for (let i = 0; i < N; i++) {
    time.push(Number(input[iter++]));
  }

  const Min = Math.min(...time);

  let start = BigInt(Min);
  let end = BigInt(Min * M);
  let answer = end;
  while (start <= end) {
    const mid = BigInt((start + end) / 2n);
    let now = 0n;
    for (let i = 0; i < N; i++) {
      now += mid / BigInt(time[i]);
    }
    if (now >= M) {
      if (answer > mid) answer = mid;
      end = mid - 1n;
    } else start = mid + 1n;
  }

  console.log(String(answer));
  process.exit();
});
